core/sql/arkcmp/CompException.cpp

```diff
@@ -113,11 +114,21 @@ FatalException::FatalException(const char *msg,
   else {
     msg_[0] = 0;
   }
+  if(stackTrace) {
+    strncpy(stackTrace_, stackTrace, sizeof(stackTrace_));
+    stackTrace_[sizeof(stackTrace_)-1] = 0;
```

Comment:
@prashanth-vasudev What I meant is if the stackTrace_ is declared as char *, then sizeof(stackTrace_) would return 4 or 8 bytes irrespective of the buffer size that was allocated to hold the stack trace. If it is declared as char stackTrace_[<max_stack_trace_len>], then sizeof(stackTrace_) will be correct. 

This article illustrates the concern I mentioned via a program. 
http://www.geeksforgeeks.org/using-sizof-operator-with-array-paratmeters/

This may not be applicable here though because stackTrace_ is the member variable of the class and this is the method of the class.